src/sql/workbench/contrib/notebook/browser/notebookViews/notebookViewsGrid.component.ts

```diff
@@ -0,0 +1,128 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import 'vs/css!./notebookViewsGrid';
+import { Component, OnInit, ViewChildren, QueryList, Input, Inject, forwardRef, ChangeDetectorRef } from '@angular/core';
+import { NotebookViewsCardComponent } from 'sql/workbench/contrib/notebook/browser/notebookViews/notebookViewsCard.component';
+import { ICellModel } from 'sql/workbench/services/notebook/browser/models/modelInterfaces';
+import { NotebookModel } from 'sql/workbench/services/notebook/browser/models/notebookModel';
+import { GridStack, GridStackEvent, GridStackNode } from 'gridstack';
+import 'gridstack/dist/h5/gridstack-dd-native';
+import { localize } from 'vs/nls';
+import { NotebookViewsExtension } from 'sql/workbench/services/notebook/browser/notebookViews/notebookViewsExtension';
+import { CellChangeEvent, INotebookViewCell } from 'sql/workbench/services/notebook/browser/notebookViews/notebookViews';
+
+@Component({
+	selector: 'notebook-views-grid-component',
+	templateUrl: decodeURI(require.toUrl('./notebookViewsGrid.component.html'))
+})
+export class NotebookViewsGridComponent implements OnInit {
+	@Input() cells: ICellModel[];
+	@Input() model: NotebookModel;
+	@Input() views: NotebookViewsExtension;
+
+	@ViewChildren(NotebookViewsCardComponent) private _items: QueryList<NotebookViewsCardComponent>;
+
+	protected _grid: GridStack;
+	public loaded: boolean;
+
+	constructor(
+		@Inject(forwardRef(() => ChangeDetectorRef)) private _changeRef: ChangeDetectorRef,
+	) {
+		this.loaded = false;
+	}
+
+	public get empty(): boolean {
+		return !this._items || !this._items.find(item => item.display);
+	}
+
+	public get hiddenItems(): NotebookViewsCardComponent[] {
+		return this._items.filter(item => !item.display);
+	}
+
+	public get emptyText(): String {
+		return localize('emptyText', "This view is empty. Add a cell to this view by clicking the Insert Cells button.");
+	}
+
+	ngOnInit() {
+		const self = this;
+
+		setTimeout(() => {
```

Comment:
Comment explaining why there's a timeout here

Reply:
Removed the timeout as it was unnecessary.  Moved the code to ngAfterViewInit.